pallets/check-membership/src/loose/mod.rs

```diff
@@ -0,0 +1,64 @@
+//! Pallet that demonstrates a minimal access control check. When a user calls this pallet's
+//! only dispatchable function, `check_membership`, the caller is checked against a set of approved
+//! callers. Only if the caller is approved, do they successfully emit the event.
```

Comment:
```suggestion
//! callers. If the caller is a member of the set, the pallet's `IsAMember` event is emitted. Otherwise a `NotAMember` error is returned.
```

Reply:
I agree with your rewording; thanks.

> Is it ok for callers to assume that "no error" means "is member"?

If a non-member calls, there will be a `NotAMember` error. So it is also true that no error means the caller is a member. Those two statements are contrapositives of each other.

I agree that in a more real-world example, after the membership check is complete, the function would go on to do something more useful than just emit an event. And part of "something more useful" would likely be checking for additional error conditions. But it would still be the case that a non-member call would always cause an error.

What kind of warning were you thinking of?

Reply:
So in "normal" rust the `Result` type encapsulates both success and failure and this is idiomatic:

```
match beef() {
  Ok(steak) => season(steak),
  Err(e) => sound_cattle_escaped_alarm(e),
}
```

What happens here is slightly different though (or I'm reading it wrong): the return value only contains the error part and on success I get `Ok(())`. To use the successful outcome of my call I need to check the event.

If that is correct then it's something that could trip up learners. In particular they might end up doing:

```
maybe_steak = beef();
if maybe_steak.is_ok() {
  // not error => assume there's steak to be had
  let steak = retrieve_the_correct_event();
  season(steak);
}
```

Now that I type it out though I'm not so sure there's anything we need to add to your text. At the end of the day an `Ok(())` means all is good and that I can go dig up my event and do what I need to do with the result.